addons/hr_contract/tests/test_contract.py

```diff
@@ -116,3 +117,60 @@ def test_current_contract_stage_change(self):
 
         draft_contract.state = 'draft'
         self.assertEqual(self.employee.contract_id, contract)
+
+    def test_check_multi_company_contract_expiration(self):
+        """
+           Check that the contract expiration based on the res setting.
+
+           Test flow:
+            - Set contract end day and work permit end days in the main company
+            - Create a John Doe employee in the main company
+            - Create a John Doe's contract
+            - Create a TestCompany2 company and set contract end days and work permit end days
+            - Create a John Smith employee in the TestCompany2 company
+            - Create a John Smith's contract
+            - Run automated actions (HR Contract: update state)
+            - Check if the expiration activity is scheduled or not
+            - A few days after run automated actions (HR Contract: update state)
+            - Again check if the expiration activity is scheduled or not
+        """
+
+        main_company = self.env.ref('base.main_company')
+        main_company.contract_end_days = 10
+        main_company.work_permit_end_days = 10
+
+        employee = self.env['hr.employee'].create({
+            'name': 'John Doe',
+            'visa_expire': date(2023, 1, 1) + relativedelta(days=10),
+            'company_id': main_company.id,
+        })
+
+        contract = self.create_contract('open', 'normal', date(2023, 1, 1), date(2023, 1, 10), employee.id)
+
+        company_2 = self.env['res.company'].create({
+            'name': 'TestCompany2',
+            'country_id': self.env.ref('base.be').id,
+            'contract_end_days' : 7,
+            'work_permit_end_days': 10,
+        })
+
+        employee1 = self.env['hr.employee'].create({
+            'name': 'Jane Smith',
+            'visa_expire': date(2023, 1, 1) + relativedelta(days=15),
+            'company_id': company_2.id,
+        })
+        contract1 = self.create_contract('open', 'normal', date(2023, 1, 1), date(2023, 1, 20), employee1.id)
+
+        with freeze_time('2023-01-01'):
+            self.env['hr.contract'].update_state()
+
+            mail_activity = self.env['mail.activity'].search([('res_id', '=', contract.id), ('res_model', '=', 'hr.contract')])
+            self.assertTrue(mail_activity.exists(), "There should be reminder activity as employee contract going to end soon")
+
+            mail_activity2 = self.env['mail.activity'].search([('res_id', '=', contract1.id), ('res_model', '=', 'hr.contract')])
+            self.assertFalse(mail_activity2.exists(), "There should be reminder activity as employee contract going to end soon")
```

Comment:
In this case, Shouldn't the error text read something like "There should be no reminder as the contract is not yet about to expire."?